docs/DeveloperGuide.md

```diff
@@ -329,11 +329,50 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `17` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+**Performance Requirements**
+- P1: The system should be able to hold up to 1000 persons without noticeable sluggishness in performance for typical usage.
+- P2: The application should respond within two seconds for most user operations.
+
+**Usability Requirements**
+- U1: A user with above-average typing speed for regular English text (not code, not system admin commands) should be able to accomplish most tasks faster using keyboard commands than using the mouse.
+- U2: The application should target users who can type fast and prefer typing over other means of input, such as clicking buttons, selecting dropdowns and drag-and-drop means.
+- U3: The GUI should provide clear and user-friendly error messages when operations fail to assist users in correcting mistakes.
+- U4: The GUI should not cause any resolution-related inconveniences for standard screen resolutions of 1920x1080 and higher, and for screen scales of 100% and 125%.
+- U5: The GUI should be usable for resolutions of 1280x720 and higher, and for screen scales of 150%.
+
+**Compatibility Requirements**
+- C1: The application should work on any _mainstream OS_ as long as it has Java `17` or above installed.
+- C2: The application should be platform-independent and work on Windows, Linux, and OS-X without relying on OS-dependent libraries or features.
```

Comment:
What's the difference between C1 and C2 here? Maybe can just put them together?

Reply:
I think C1 is emphasising more that Java 17 is required, while C2 is emphasising that the app shouldn't use OS-dependent libraries.